src/Contact.php

```diff
@@ -29,80 +24,31 @@
     /** @var string */
     private $email;
 
-    public function __construct($data = [])
-    {
-        $this->merge($data);
-    }
-
-    protected function doMerge($data, $overwrite = false)
-    {
-        MergeHelper::mergeFields($this->name, $data['name'] ?? null, $overwrite);
-        MergeHelper::mergeFields($this->url, $data['url'] ?? null, $overwrite);
-        MergeHelper::mergeFields($this->email, $data['email'] ?? null, $overwrite);
-
-        $this->mergeExtensions($data, $overwrite);
-    }
-
-    protected function doExport(): array
-    {
-        return [
-            'name' => $this->name,
-            'url' => $this->url,
-            'email' => $this->email,
-        ];
-    }
-
-    /**
-     * @return string
-     */
-    public function getName()
-    {
-        return $this->name;
-    }
-
-    /**
-     * @param string $name
-     */
-    public function setName($name): self
+    public function __construct(array $data = [])
     {
-        $this->name = $name;
+        $this->name = $data['name'] ?? null;
+        $this->url = $data['url'] ?? null;
+        $this->email = $data['email'] ?? null;
 
-        return $this;
+        $this->mergeExtensions($data);
     }
 
-    /**
-     * @return string
-     */
-    public function getUrl()
+    protected function export(): array
     {
-        return $this->url;
-    }
+        $return = [];
 
-    /**
-     * @param string $url
-     */
-    public function setUrl($url): self
-    {
-        $this->url = $url;
+        if ($this->name) {
```

Comment:
Shouldn't we check for null explicitely since the empty string also pass this check ?

Reply:
sure

Reply:
As I said it lacks validation for now, if we don't allow empty string we should fail on constructor (or setter), now during exporting, but as far as swagger editor validator concerns empty string here is OK

Reply:
Sorry I wasn't very clear, I mean we should use `null !==` because if the empty string is used, it won't be exported (see https://3v4l.org/GcHem).